diesel/src/query_builder/select_by_statement/dsl_impls.rs

```diff
@@ -0,0 +1,362 @@
+
+    fn with_lock(self, lock: Lock) -> Self::Output {
+        SelectByStatement::new(self.inner.with_lock(lock))
+    }
+}
+
+impl<ST, S, STMT, Modifier> ModifyLockDsl<Modifier>
+    for SelectByStatement<S, STMT>
+where
+    Self: SelectQuery,
+    STMT: ModifyLockDsl<Modifier>,
+    SelectByStatement<S, STMT::Output>: SelectQuery<SqlType = ST>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn modify_lock(self, modifier: Modifier) -> Self::Output {
+        SelectByStatement::new(self.inner.modify_lock(modifier))
+    }
+}
+
+// impl<'a, F, S, D, W, O, L, Of, G, DB> BoxedDsl<'a, DB>
+//     for SelectStatement<F, SelectClause<S>, D, W, O, L, Of, G>
+// where
+//     Self: AsQuery,
+//     DB: Backend,
+//     S: QueryFragment<DB> + SelectableExpression<F> + Send + 'a,
+//     D: QueryFragment<DB> + Send + 'a,
+//     W: Into<BoxedWhereClause<'a, DB>>,
+//     O: Into<Option<Box<dyn QueryFragment<DB> + Send + 'a>>>,
+//     L: QueryFragment<DB> + Send + 'a,
+//     Of: QueryFragment<DB> + Send + 'a,
+//     G: QueryFragment<DB> + Send + 'a,
+// {
+//     type Output = BoxedSelectStatement<'a, S::SqlType, F, DB>;
+
+//     fn internal_into_boxed(self) -> Self::Output {
+//         BoxedSelectStatement::new(
+//             Box::new(self.select.0),
+//             self.from,
+//             Box::new(self.distinct),
+//             self.where_clause.into(),
+//             self.order.into(),
+//             Box::new(self.limit),
+//             Box::new(self.offset),
+//             Box::new(self.group_by),
+//         )
+//     }
+// }
+
+// impl<'a, F, D, W, O, L, Of, G, DB> BoxedDsl<'a, DB>
+//     for SelectStatement<F, DefaultSelectClause, D, W, O, L, Of, G>
+// where
+//     Self: AsQuery,
+//     DB: Backend,
+//     F: QuerySource,
+//     F::DefaultSelection: QueryFragment<DB> + Send + 'a,
+//     D: QueryFragment<DB> + Send + 'a,
+//     W: Into<BoxedWhereClause<'a, DB>>,
+//     O: Into<Option<Box<dyn QueryFragment<DB> + Send + 'a>>>,
+//     L: QueryFragment<DB> + Send + 'a,
+//     Of: QueryFragment<DB> + Send + 'a,
+//     G: QueryFragment<DB> + Send + 'a,
+// {
+//     type Output = BoxedSelectStatement<'a, <F::DefaultSelection as Expression>::SqlType, F, DB>;
+//     fn internal_into_boxed(self) -> Self::Output {
+//         BoxedSelectStatement::new(
+//             Box::new(self.from.default_selection()),
+//             self.from,
+//             Box::new(self.distinct),
+//             self.where_clause.into(),
+//             self.order.into(),
+//             Box::new(self.limit),
+//             Box::new(self.offset),
+//             Box::new(self.group_by),
+//         )
+//     }
+// }
+
+impl<S, STMT> HasTable for SelectByStatement<S, STMT>
+where
+    STMT: HasTable,
+{
+    type Table = STMT::Table;
+
+    fn table() -> Self::Table {
+        STMT::table()
+    }
+}
+
+// no IntoUpdateTarget: if you'd like to `.into_update_target`, you should not first `.select_by`
```

Comment:
Should we implement `IntoUpdateTarget`

Reply:
No, we do only implement `IntoUpdateTarget` for cases without explicit select clause to allow using it as "where clause" in update/delete statements.